src/sparseml/transformers/utils/preprocesing_funtions.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def evolved_codealpaca_dataset(data):
+    PROMPT_DICT = """[Instructions]:\n{instruction}\n\n[Response]:"""
+    data["text"] = PROMPT_DICT.format_map(data)
+
+    return data
```

Comment:
Is this preprocessing function only applicable to this one dataset? Why do we need to create this separate feature instead of just adding the dataset to the TextGenerationDataset registry so we can call it as a string?

Reply:
EDIT: after talking with Abhinav, he is using a custom split of the evolcodealpaca dataset, so going through the dataset in our registry will not work. I think we should still refactor the preprocessing registry into a real RegistryMixIn, but I understand the need for it now